### DISCUSSION TITLE
[Request] Sort functionality in Pivot Table

### DISCUSSION THREAD

**Discussion by lctdulac** [apache/superset#18469] [Ideas]
Hi SuperSet community.
The Sort functionality in Pivot table seem to be lacking. I think this would be an interesting feature when creating pivot table with ordinal features.

An example :

<img width="694" alt="Screenshot 2021-03-25 at 11 42 43" src="https://user-images.githubusercontent.com/46674763/112462006-dbdcf580-8d60-11eb-912f-a6197b89469b.png">

Here it would be amazing to be able to sort both level 1 and level 2 of this table, as we would do in SQL.

In particular it would be great to be able to sort like this: 
Level 1: sorted alphabetically
Level 2: sorted with custom SQL like: Trusted > Neutral > Low > Medium > High

What do you guys think ?

**Comment by junlincc** [comment]
@lctdulac we have a new pivot table available soon. https://github.com/apache-superset/superset-ui/pull/1023
thanks to @kgabryje 
I'm not sure we can support sorting to this extent though.

**Comment by lctdulac** [comment]
@junlincc thanks! the new pivot table looks great. I still think it would be amazing to be able to sort at least one level for ordinal data.

**Comment by kamalkeshavani-aiinside** [comment]
@junlincc 
I just checked the new Pivot Table v2 viz, it's a great improvement.
About the sorting, currently it supports sorting the rows and columns based on keys or values.
In case of values, by default it considers right most column values or the last(bottom) row values, so it would be great if we can have a 'Sort By' option in Query section to decide what value to be used for sorting both rows and columns.

**Reply by rtotheb2000** [reply]
+1
